source/features/comment-shortcuts.tsx

```diff
@@ -0,0 +1,68 @@
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import {isEditable} from '../helpers/dom-utils';
+
+const shortcutClass = new Map<string, string>([
+	['e', '.unminimized-comment [aria-label="Edit comment"]:not([hidden])'],
+	['d', '[aria-label="Delete comment"]']
+]);
+
+function runShortcuts(event: KeyboardEvent): void {
+	if (isEditable(event.target)) {
+		return;
+	}
+
+	const focusedComment = select(':target')!;
+
+	if (['j', 'k'].includes(event.key)) {
+		event.preventDefault();
+
+		const items = select.all('.js-minimizable-comment-group')
+			.filter(comment => !comment.querySelector('.minimized-comment:not(.d-none)'));
+		// `j` goes to the next comment, `k` goes back a comment
+		const direction = event.key === 'j' ? 1 : -1;
+
+		const currentIndex = items.indexOf(focusedComment);
+
+		// Start at 0 if nothing is; clamp index
+		const chosenCommentIndex = Math.min(
+			Math.max(0, currentIndex + direction),
+			items.length - 1
+		);
+
+		// Focus comment without pushing to history
+		location.replace('#' + items[chosenCommentIndex].id);
+		items[chosenCommentIndex].scrollIntoView();
```

Comment:
Can you make a gif showing the difference between having this and not having this? `location.replace` seems to already to it.

Reply:
Related: GitHub adds an annoying and buggy double-scroll that happens every time a comment is focused; this is unrelated to us and doesn't seem to be solvable (but I wish it could be)

![](https://user-images.githubusercontent.com/1402241/85280422-67e78700-b488-11ea-998c-c8dc909ddae0.gif)

Reply:
I am not good at these things hopefully you can tell from the gif.

the scroll to void avoids a second jump.

![Video_2020-06-22_081805](https://user-images.githubusercontent.com/16872793/85286744-34ddcd00-b461-11ea-8473-c2afaaf9e603.gif)

Reply:
That issue appears regardless of that line in my experience (Chrome)

Reply:
Me too but there a second jump (That I noticed) without the scroll to view.

I have no issue to removing it

Reply:
Can be removed